File: LeetCode_Marathons/LCM6/Medium/Count_Number_of_Homogenous_Substrings/a.py
class Solution:
    def countHomogenous(self, s: str) -> int:
        mod: int = 1000000007
        ans, keyCounter = 0, 0
        s += "*"
        key = "*"

        for myCh in s:
            if myCh == key: keyCounter += 1
            else:
                ansForKey: int
                ansForKey = (keyCounter + 1) * keyCounter // 2
                ans = (ans + ansForKey) % mod
                keyCounter = 1
                key = myCh
        
        return ans

File: LeetCode_Marathons/LCM6/Medium/Count_Number_of_Homogenous_Substrings/a.java
class Solution {
    public int countHomogenous(String s) {
        long mod = 1000000007;
        long ans = 0;
        Character key = '*';
        s += "*";
        long keyCounter = 0;
        for (Character myCh : s.toCharArray()) {
            if (myCh == key) keyCounter++;
            else {
                long ansForKey;
                ansForKey = (keyCounter + 1) * keyCounter / 2;
                ans = (ans + ansForKey) % mod;
                keyCounter = 1;
                key = myCh;
            }
        }

        return (int) ans;
    }
}